### PR TITLE
Dgs 174

### PR DESCRIPTION
**Pull request by ErnestasLuza**
Screenshot from 2022-07-19 10-53-43
image
image
image

- Tooltip for Shippment settings Automated parcel return
- Download return parcel label buttons in Order view list and below form
- Print return label in Order view
- Bulk action if Automated settings are on print Return and Shipment labels.

### CODE REVIEW

File: controllers/admin/AdminDPDBalticsShipmentSettingsController.php [hunk 1]
```diff
@@ -187,6 +187,13 @@ protected function initOptions()
                         'cast' => 'intval',
                         'type' => 'bool',
                     ],
+                    Config::AUTOMATED_PARCEL_RETURN => [
+                        'title' => $this->l('Automated parcel return'),
```
Comment (arturas-bes) [inline controllers/admin/AdminDPDBalticsShipmentSettingsController.php:191]: Maybe name could be something like Automatically generate return label P.S Translations has to be added aswell

File: controllers/front/ShipmentReturn.php [hunk 2]
```diff
@@ -20,11 +20,17 @@ public function postProcess()
 
         $orderid = Tools::getValue('id_order');
         $returnTemplateId = Tools::getValue('return_template_id');
+
+        if (!$returnTemplateId) {
```
Comment (arturas-bes) [inline controllers/front/ShipmentReturn.php:24]: Why this is needed?

File: controllers/front/ShipmentReturn.php [hunk 3]
```diff
@@ -20,11 +20,17 @@ public function postProcess()
 
         $orderid = Tools::getValue('id_order');
         $returnTemplateId = Tools::getValue('return_template_id');
```
Comment (arturas-bes) [inline controllers/front/ShipmentReturn.php:22]:
```suggestion
        $returnTemplateId = Tools::getValue('return_template_id') ?: 1; In what cases it is not returned?
```

Comment (arturas-bes) [reply controllers/front/ShipmentReturn.php:22]: Also constant should be used random numbers in code is a bad practice

File: dpdbaltics.php [hunk 4]
```diff
@@ -1086,10 +1103,15 @@ public function printLabel($idShipment)
         $shipment = new DPDShipment($idShipment);
         $format = $shipment->label_format;
         $position = $shipment->label_position;
-
+        $isAutomated = Configuration::get(Config::AUTOMATED_PARCEL_RETURN);
         try {
             /** @var ParcelPrintResponse $parcelPrintResponse */
-            $parcelPrintResponse = $labelApiService->printLabel($shipment->pl_number, $format, $position, false);
+            if ($isAutomated) {
```
Comment (arturas-bes) [inline dpdbaltics.php:1109]: I would set this logic to new service or atleast private function so it is easer to read

File: views/templates/hook/admin/admin-order.tpl [hunk 5]
```diff
@@ -66,6 +67,15 @@
                                         {/if}
                                     </a>
                                 </td>
+                                {if !$isAutomated}
+                                <td>
+                                    <form id="order-return-form" action="{$href}" method="post">
+                                        <button class="btn btn-link pt-0 pb-0" type="submit">
+                                            {l s='Download return label' mod='dpdbaltics'}
```
Comment (arturas-bes) [inline views/templates/hook/admin/admin-order.tpl:74]: Please format return label view button in a same way as it is with shipment buttpn

**Review by arturas-bes — CHANGES_REQUESTED** [review]

**Comment by arturas-bes** [comment]
In BO list, the button seems out of style, please change it

**Comment by arturas-bes** [comment]
In the bottom view return  button is bellow, could you do it side by side? we have space up there

**Comment by arturas-bes** [comment]
Cannot see, how this return logic is handled in older prestashop versions, please check code on the first compatible prestashop 1.7 version.

File: controllers/admin/AdminDPDBalticsShipmentSettingsController.php (hunk 1 above)
Comment (ErnestasLuza) [reply controllers/admin/AdminDPDBalticsShipmentSettingsController.php]: Done

**Comment by ErnestasLuza** [comment]
Fixed buttons, and moved the button below the form to the side. Tested DPD module with this functionality on PS 1.7.1 and 1.7.6 works as intended.

**Review by arturas-bes — APPROVED** [review]

**Pull request merged**